pkg/apis/configuration/v1/types.go

```diff
@@ -652,82 +880,128 @@ type VariableCondition struct {
-	Scope                 string   `json:"scope"`
-	RedirectURI           string   `json:"redirectURI"`
-	EndSessionEndpoint    string   `json:"endSessionEndpoint"`
-	PostLogoutRedirectURI string   `json:"postLogoutRedirectURI"`
-	ZoneSyncLeeway        *int     `json:"zoneSyncLeeway"`
-	AuthExtraArgs         []string `json:"authExtraArgs"`
-	AccessTokenEnable     bool     `json:"accessTokenEnable"`
-	PKCEEnable            bool     `json:"pkceEnable"`
-}
-
-// WAF defines an WAF policy.
+	// URL for the authorization endpoint provided by your OpenID Connect provider.
+	AuthEndpoint string `json:"authEndpoint"`
+	// URL for the token endpoint provided by your OpenID Connect provider.
+	TokenEndpoint string `json:"tokenEndpoint"`
+	// URL for the JSON Web Key Set (JWK) document provided by your OpenID Connect provider.
+	JWKSURI string `json:"jwksURI"`
+	// The client ID provided by your OpenID Connect provider.
+	ClientID string `json:"clientID"`
+	// The name of the Kubernetes secret that stores the client secret provided by your OpenID Connect provider. It must be in the same namespace as the Policy resource. The secret must be of the type nginx.org/oidc, and the secret under the key client-secret, otherwise the secret will be rejected as invalid. If PKCE is enabled, this should be not configured.
+	ClientSecret string `json:"clientSecret"`
+	// List of OpenID Connect scopes. The scope openid always needs to be present and others can be added concatenating them with a + sign, for example openid+profile+email, openid+email+userDefinedScope. The default is openid.
+	Scope string `json:"scope"`
+	// Allows overriding the default redirect URI. The default is /_codexch.
+	RedirectURI string `json:"redirectURI"`
+	// URL provided by your OpenID Connect provider to request the end user be logged out.
+	EndSessionEndpoint string `json:"endSessionEndpoint"`
+	// URI to redirect to after the logout has been performed. Requires endSessionEndpoint. The default is /_logout.
+	PostLogoutRedirectURI string `json:"postLogoutRedirectURI"`
+	// Specifies the maximum timeout in milliseconds for synchronizing ID/access tokens and shared values between Ingress Controller pods. The default is 200.
+	ZoneSyncLeeway *int `json:"zoneSyncLeeway"`
+	// A list of extra URL arguments to pass to the authorization endpoint provided by your OpenID Connect provider. Arguments must be URL encoded, multiple arguments may be included in the list, for example [ arg1=value1, arg2=value2 ]
+	AuthExtraArgs []string `json:"authExtraArgs"`
+	// Option of whether Bearer token is used to authorize NGINX to access protected backend.
+	AccessTokenEnable bool `json:"accessTokenEnable"`
+	// Switches Proof Key for Code Exchange on. The OpenID client needs to be in public mode. clientSecret is not used in this mode.
+	PKCEEnable bool `json:"pkceEnable"`
+}
+
+// The WAF policy configures NGINX Plus to secure client requests using App Protect WAF policies.
 type WAF struct {
```

Comment:
Do we want to change the WAF CRDs?

Reply:
@haywoodsh I think this one is from the the NIC policy, this task is only for 5 major NIC CRDs